tasks/dotnetcore/env/versions.go

```diff
@@ -68,22 +79,98 @@ func (t DotNetCoreEnvVersions) Execute(options tasks.Options, upstream map[strin
 	}
 }
 
-func checkVersions(envVars map[string]string) ([]string, string) {
-	errorMessage := "Unable to complete this health check because we ran into some unexpected errors when attempting to collect this application's .NET Core SDK version:\n"
+func (t DotNetCoreEnvVersions) checkVersions(envVars map[string]string) ([]string, string) {
+	source := 0
+	errorMessages := ""
+	versions, errMessage, keepGoing := t.checkVersionWithSource(envVars, source)
+	for keepGoing {
+		source++
+		errorMessages += errMessage
+		versions, errMessage, keepGoing = t.checkVersionWithSource(envVars, source)
+	}
```

Comment:
The only thing I might change is calling this function outside of the function and inside it.  Maybe something like:
keepGoing := true
for keepGoing{
   versions, errMessage, keepGoing := t.checkVersionWithSource(envVars, source)
   errorMessages += errMessage
   source++
}

This isn't necessary to change it just makes more sense when I read the code

Reply:
Good call out! I've refactored it to be more readable (I hope)